x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockUtils.java

```diff
@@ -207,13 +212,23 @@ public static void appendValue(Block.Builder builder, Object val, ElementType ty
         switch (type) {
             case LONG -> ((LongBlock.Builder) builder).appendLong((Long) val);
             case INT -> ((IntBlock.Builder) builder).appendInt((Integer) val);
-            case BYTES_REF -> ((BytesRefBlock.Builder) builder).appendBytesRef(toBytesRef(val));
+            case BYTES_REF -> ((BytesRefBlock.Builder) builder).appendBytesRef(spatialToBytesRef(val));
             case DOUBLE -> ((DoubleBlock.Builder) builder).appendDouble((Double) val);
             case BOOLEAN -> ((BooleanBlock.Builder) builder).appendBoolean((Boolean) val);
             default -> throw new UnsupportedOperationException("unsupported element type [" + type + "]");
         }
     }
 
+    private static BytesRef spatialToBytesRef(Object val) {
```

Comment:
Probably worth a rename and a javadoc. I'd love the name to imply that it also converts strings to bytes refs.

Reply:
Not sure a good name. `allJavaTypesToBytesRef`. Or, even, `toBytesRef` is an ok name, I guess. But I'm just bike shedding.

Reply:
I felt like `toBytesRef` would clash with the previous existing method, so I wanted a new name and `toBytesRefAlsoConsideringSpatial` seemed even worse.

Reply:
I went for `toBytesRef`, we can all follow the stack anyway.

Reply:
This is now entirely removed, as it was only really needed for tests that themselves were not correctly written.